ResearchKit/Onboarding/ORKRegistrationStep.m

```diff
@@ -0,0 +1,258 @@
+        [formItems addObject:passwordItem];
+        
+        NSString *originalItemIdentifier = [passwordItem.identifier copy];
+        ORKConfirmTextAnswerFormat *confirmAnswerFormat = [[ORKConfirmTextAnswerFormat alloc]
+                                                           initWithOriginalItemIdentifier:originalItemIdentifier];
+        confirmAnswerFormat.multipleLines = NO;
+        confirmAnswerFormat.secureTextEntry = YES;
+        confirmAnswerFormat.spellCheckingType = UITextSpellCheckingTypeNo;
+        confirmAnswerFormat.autocapitalizationType = UITextAutocapitalizationTypeNone;
+        confirmAnswerFormat.autocorrectionType = UITextAutocorrectionTypeNo;
+
+        ORKFormItem *confirmItem = [[ORKFormItem alloc] initWithIdentifier:ORKRegistrationFormItemConfirmPassword
+                                                                      text:ORKLocalizedString(@"CONFIRM_PASSWORD_FORM_ITEM_TITLE", nil)
+                                                               answerFormat:confirmAnswerFormat];
+        confirmItem.placeholder = ORKLocalizedString(@"CONFIRM_PASSWORD_FORM_ITEM_PLACEHOLDER", nil);
+        confirmItem.optional = NO;
+        
+        [formItems addObject:confirmItem];
+    }
+
+    if (!(_options & ORKRegistrationStepDefault)) {
+        ORKFormItem *item = [[ORKFormItem alloc] initWithSectionTitle:ORKLocalizedString(@"ADDITIONAL_INFO_SECTION_TITLE", nil)];
+        
+        [formItems addObject:item];
+    }
+    
+    if (_options & ORKRegistrationStepIncludeGivenName) {
+        ORKTextAnswerFormat *answerFormat = [ORKAnswerFormat textAnswerFormat];
+        answerFormat.multipleLines = NO;
+        
+        ORKFormItem *item = [[ORKFormItem alloc] initWithIdentifier:ORKRegistrationFormItemGivenName
+                                                               text:ORKLocalizedString(@"CONSENT_NAME_GIVEN", nil)
+                                                       answerFormat:answerFormat];
+        item.placeholder = ORKLocalizedString(@"GIVEN_NAME_ITEM_PLACEHOLDER", nil);
+        item.optional = NO;
+        
+        [formItems addObject:item];
+    }
+    
+    if (_options & ORKRegistrationStepIncludeFamilyName) {
+        ORKTextAnswerFormat *answerFormat = [ORKAnswerFormat textAnswerFormat];
+        answerFormat.multipleLines = NO;
+        
+        ORKFormItem *item = [[ORKFormItem alloc] initWithIdentifier:ORKRegistrationFormItemFamilyName
+                                                               text:ORKLocalizedString(@"CONSENT_NAME_FAMILY", nil)
+                                                       answerFormat:answerFormat];
+        item.placeholder = ORKLocalizedString(@"FAMILY_NAME_ITEM_PLACEHOLDER", nil);
+        item.optional = NO;
+        
+        [formItems addObject:item];
+    }
+    
+    if (_options & ORKRegistrationStepIncludeGender) {
+        NSArray *textChoices = @[[ORKTextChoice choiceWithText:ORKLocalizedString(@"GENDER_MALE", nil) value:@0],
```

Comment:
Without looking at the code, it is a little hard for developer to figure out the meaning of `0/1/2` ...